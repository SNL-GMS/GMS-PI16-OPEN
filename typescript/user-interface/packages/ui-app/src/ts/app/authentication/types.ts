import { UserSessionTypes } from '@gms/ui-state';

export interface Authenticator {
  authenticateWith(userName: string): Promise<UserSessionTypes.AuthStatus>;
  checkIsAuthenticated(): Promise<UserSessionTypes.AuthStatus>;
  unAuthenticateWith(): Promise<UserSessionTypes.AuthStatus>;
  logout(setAuthStatus: (status: UserSessionTypes.AuthStatus) => void): void;
}

export interface LoginRequestParams {
  userName: string;
}
