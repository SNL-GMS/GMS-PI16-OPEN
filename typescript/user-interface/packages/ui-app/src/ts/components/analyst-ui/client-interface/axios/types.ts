/* eslint-disable @typescript-eslint/no-explicit-any */
import {
  ChannelSegmentTypes,
  EventTypes,
  QcMaskTypes,
  SignalDetectionTypes,
  WaveformTypes,
  WorkflowTypes
} from '@gms/common-model';
import { AxiosRequestConfig } from 'axios';
import { QueryResult } from 'react-query';

export interface EventMutationArgs {
  variables: {
    events: EventTypes.Event[];
  };
}

export interface QcMaskMutationArgs {
  variables: {
    qcMasks: QcMaskTypes.QcMask[];
  };
}

export interface SignalDetectionMutationArgs {
  variables: {
    signalDetections: SignalDetectionTypes.SignalDetection[];
  };
}

// ***************************************
// Workflow Interval Query and Mutation interfaces
// ***************************************
export interface StageIntervalsByStageIdAndTimeRequest {
  startTime: number;
  endTime: number;
  stageIds: WorkflowTypes.WorkflowDefinitionId[];
}
export interface UserRequest {
  userName: string;
  time: number;
}

export interface UpdateActivityIntervalStatusRequest extends UserRequest {
  activityIntervalId: WorkflowTypes.IntervalId;
  stageIntervalId: WorkflowTypes.IntervalId;
  status: WorkflowTypes.IntervalStatus;
}

export interface UpdateInteractiveAnalysisStageIntervalStatusRequest extends UserRequest {
  stageIntervalId: WorkflowTypes.IntervalId;
  status: WorkflowTypes.IntervalStatus;
}

export interface SplitActivityIntervalRequest {
  intervalId: WorkflowTypes.IntervalId;
  duration: number;
}
export interface EventQueryProps {
  eventQuery: QueryResult<EventTypes.Event[], any>;
}

export interface QcMaskQueryProps {
  qcMaskQuery: QueryResult<QcMaskTypes.QcMask[], any>;
}

export interface SignalDetectionQueryProps {
  signalDetectionQuery: QueryResult<SignalDetectionTypes.SignalDetection[], any>;
}

export interface WaveformQueryProps {
  waveformQuery: QueryResult<ChannelSegmentTypes.ChannelSegment<WaveformTypes.Waveform>[], any>;
}

/**
 * An axios request known to have a waveform query request in its data.
 */
export type WaveformAxiosRequestConfig = AxiosRequestConfig & {
  data: WaveformTypes.WaveformQueryArgs;
};
