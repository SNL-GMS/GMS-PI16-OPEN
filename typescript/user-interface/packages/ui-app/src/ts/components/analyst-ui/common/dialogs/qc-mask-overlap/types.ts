import { QcMaskTypes } from '@gms/common-model';

// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { CoordinatesPx } from '../qc-mask-form/types';
// !FIX ESLINT CIRCULAR DEPENDENCY
// eslint-disable-next-line import/no-cycle
import { QcMaskDialogBoxType } from '../types';

/**
 * QcMaskOverlap Props
 */
export interface QcMaskOverlapProps {
  masks: QcMaskTypes.QcMask[];
  // If set, any context menus opened as children will be created at given x/y coordinates
  contextMenuCoordinates?: CoordinatesPx;
  openNewContextMenu(
    x: number,
    y: number,
    mask: QcMaskTypes.QcMask,
    dialogType: QcMaskDialogBoxType
  );
  selectMask(mask: QcMaskTypes.QcMask);
}

/**
 * QcMaskOverlap  State
 */
export interface QcMaskOverlapState {
  selectedMask: QcMaskTypes.QcMask;
  createDialogOfType?: QcMaskDialogBoxType;
}
export interface QcMaskTableButtonParams {
  onClick(x: number, y: number, params: any);
}
