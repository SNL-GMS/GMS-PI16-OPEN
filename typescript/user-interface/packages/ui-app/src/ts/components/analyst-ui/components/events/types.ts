import { GQLTypes } from '@gms/common-graphql';
import { CommonTypes, EventTypes, WorkflowTypes } from '@gms/common-model';
import GoldenLayout from '@gms/golden-layout';
import { ChildProps, MutationFunction } from 'react-apollo';

/**
 * Types of events which can be show
 */
export enum EventFilters {
  EDGE = 'Edge',
  COMPLETED = 'Completed'
}

/**
 * Table row object for events
 */
export interface EventsRow {
  id: string;
  eventHypId: string;
  isOpen: boolean;
  stageId: string;
  lat: number;
  lon: number;
  depth: number;
  time: number;
  activeAnalysts: string[];
  numDetections: number;
  status: string;
  edgeEvent: boolean;
  signalDetectionConflicts: SignalDetectionConflict[];
}

export interface SignalDetectionConflict {
  stationName: string;
  id: string;
  arrivalTime: number;
  phase: CommonTypes.PhaseType;
}
/**
 * Event list local state
 */
export interface EventsState {
  currentTimeInterval: CommonTypes.TimeRange;
  suppressScrollOnNewData: boolean;
  showEventOfType: Map<EventFilters, boolean>;
}

/**
 * Mutations used in the event list
 */
export interface EventsMutations {
  updateEvents: MutationFunction;
  saveEvent: MutationFunction;
  saveAllModifiedEvents: MutationFunction;
}

/**
 * Props mapped in from Redux state
 */
export interface EventsReduxProps {
  // Passed in from golden-layout
  glContainer?: GoldenLayout.Container;
  currentTimeInterval: CommonTypes.TimeRange;
  analysisMode: WorkflowTypes.AnalysisMode;
  openEventId: string;
  selectedEventIds: string[];

  // callbacks
  setOpenEventId(
    event: EventTypes.Event | undefined,
    latestLocationSolutionSet: EventTypes.LocationSolutionSet | undefined,
    preferredLocationSolutionId: string | undefined
  ): void;
  setSelectedEventIds(ids: string[]): void;
}

/**
 * Consolidated props type for event list
 */
export type EventsProps = EventsReduxProps &
  ChildProps<EventsMutations> &
  GQLTypes.EventsInTimeRangeQueryProps &
  GQLTypes.DefaultStationsQueryProps &
  GQLTypes.SignalDetectionsByStationQueryProps &
  GQLTypes.WorkspaceStateProps;
