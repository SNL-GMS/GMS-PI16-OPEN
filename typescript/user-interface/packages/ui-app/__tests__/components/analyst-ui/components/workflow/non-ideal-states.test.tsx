import { TimeRange } from '@gms/common-model/lib/common/types';
import { OperationalTimePeriodConfiguration } from '@gms/common-model/lib/ui-configuration/types';
import { StageInterval, Workflow } from '@gms/common-model/lib/workflow/types';
import { WithNonIdealStates } from '@gms/ui-core-components';
import { createStore } from '@gms/ui-state';
import { cloneDeep } from 'apollo-utilities';
import Immutable from 'immutable';
import React from 'react';
import { QueryResult } from 'react-query';
import { Provider } from 'react-redux';

import {
  workflowIntervalQueryNonIdealStates,
  workflowQueryNonIdealStates
} from '../../../../../src/ts/components/analyst-ui/components/workflow/non-ideal-states';
import { WorkflowPanelProps } from '../../../../../src/ts/components/analyst-ui/components/workflow/types';
import { WorkflowPanel } from '../../../../../src/ts/components/analyst-ui/components/workflow/workflow-panel';
import { BaseDisplay } from '../../../../../src/ts/components/common-ui/components/base-display';
import { reactQueryResult } from '../../../../__data__/test-util';
import { glContainer } from './gl-container';

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
const Enzyme = require('enzyme');

const store = createStore();

const intervalQuery: QueryResult<Immutable.Map<string, StageInterval[]>, unknown> = cloneDeep(
  reactQueryResult
);

const workflowQuery: QueryResult<Workflow, unknown> = cloneDeep(reactQueryResult);
const operationalTimePeriodConfigurationQuery: QueryResult<
  OperationalTimePeriodConfiguration,
  any
> = cloneDeep(reactQueryResult);
const timeRange: TimeRange = {
  startTimeSecs: 1000,
  endTimeSecs: 2000
};
describe('workflow non-ideal-states', () => {
  it('workflow query exists', () => {
    expect(workflowQueryNonIdealStates).toBeDefined();
  });

  it('workflow query error matches snapshot', () => {
    const NonIdealElement = WithNonIdealStates<WorkflowPanelProps>(
      [...workflowQueryNonIdealStates],
      WorkflowPanel
    );
    workflowQuery.isError = true;
    workflowQuery.isLoading = false;
    intervalQuery.isError = false;
    intervalQuery.isLoading = false;
    operationalTimePeriodConfigurationQuery.isLoading = false;
    operationalTimePeriodConfigurationQuery.isError = false;
    const component = Enzyme.mount(
      <BaseDisplay glContainer={glContainer}>
        <Provider store={store}>
          <NonIdealElement
            workflowQuery={workflowQuery}
            workflowIntervalQuery={intervalQuery}
            operationalTimePeriodConfigurationQuery={operationalTimePeriodConfigurationQuery}
            timeRange={timeRange}
          />
        </Provider>
      </BaseDisplay>
    );
    expect(component).toMatchSnapshot();
  });

  it('workflow query loading matches snapshot', () => {
    const NonIdealElement = WithNonIdealStates<WorkflowPanelProps>(
      [...workflowQueryNonIdealStates],
      WorkflowPanel
    );
    workflowQuery.isError = false;
    workflowQuery.isLoading = true;
    intervalQuery.isError = false;
    intervalQuery.isLoading = false;
    operationalTimePeriodConfigurationQuery.isLoading = false;
    operationalTimePeriodConfigurationQuery.isError = false;
    const component = Enzyme.mount(
      <BaseDisplay glContainer={glContainer}>
        <Provider store={store}>
          <NonIdealElement
            workflowQuery={workflowQuery}
            workflowIntervalQuery={intervalQuery}
            operationalTimePeriodConfigurationQuery={operationalTimePeriodConfigurationQuery}
            timeRange={timeRange}
          />
        </Provider>
      </BaseDisplay>
    );
    expect(component).toMatchSnapshot();
  });

  it('workflow interval query empty matches snapshot', () => {
    const NonIdealElement = WithNonIdealStates<WorkflowPanelProps>(
      [...workflowIntervalQueryNonIdealStates],
      WorkflowPanel
    );
    workflowQuery.isError = false;
    workflowQuery.isLoading = false;
    intervalQuery.isError = false;
    intervalQuery.isLoading = false;
    intervalQuery.data = undefined;
    operationalTimePeriodConfigurationQuery.isLoading = false;
    operationalTimePeriodConfigurationQuery.isError = false;
    const component = Enzyme.mount(
      <BaseDisplay glContainer={glContainer}>
        <Provider store={store}>
          <NonIdealElement
            workflowQuery={workflowQuery}
            workflowIntervalQuery={intervalQuery}
            operationalTimePeriodConfigurationQuery={operationalTimePeriodConfigurationQuery}
            timeRange={timeRange}
          />
        </Provider>
      </BaseDisplay>
    );
    expect(component).toMatchSnapshot();
  });

  it('workflow interval query error matches snapshot', () => {
    const NonIdealElement = WithNonIdealStates<WorkflowPanelProps>(
      [...workflowIntervalQueryNonIdealStates],
      WorkflowPanel
    );
    workflowQuery.isError = false;
    workflowQuery.isLoading = false;
    intervalQuery.isError = true;
    intervalQuery.isLoading = false;
    operationalTimePeriodConfigurationQuery.isLoading = false;
    operationalTimePeriodConfigurationQuery.isError = false;
    const component = Enzyme.mount(
      <BaseDisplay glContainer={glContainer}>
        <Provider store={store}>
          <NonIdealElement
            workflowQuery={workflowQuery}
            workflowIntervalQuery={intervalQuery}
            operationalTimePeriodConfigurationQuery={operationalTimePeriodConfigurationQuery}
            timeRange={timeRange}
          />
        </Provider>
      </BaseDisplay>
    );
    expect(component).toMatchSnapshot();
  });

  it('workflow interval query loading matches snapshot', () => {
    const NonIdealElement = WithNonIdealStates<WorkflowPanelProps>(
      [...workflowIntervalQueryNonIdealStates],
      WorkflowPanel
    );
    workflowQuery.isError = false;
    workflowQuery.isLoading = false;
    intervalQuery.isError = false;
    intervalQuery.isLoading = true;
    operationalTimePeriodConfigurationQuery.isLoading = false;
    operationalTimePeriodConfigurationQuery.isError = false;
    const component = Enzyme.mount(
      <BaseDisplay glContainer={glContainer}>
        <Provider store={store}>
          <NonIdealElement
            workflowQuery={workflowQuery}
            workflowIntervalQuery={intervalQuery}
            operationalTimePeriodConfigurationQuery={operationalTimePeriodConfigurationQuery}
            timeRange={timeRange}
          />
        </Provider>
      </BaseDisplay>
    );
    expect(component).toMatchSnapshot();
  });

  it('workflow interval query exists', () => {
    expect(workflowIntervalQueryNonIdealStates).toBeDefined();
  });

  it('workflow interval query matches snapshot', () => {
    expect(workflowIntervalQueryNonIdealStates).toBeDefined();
  });
});
