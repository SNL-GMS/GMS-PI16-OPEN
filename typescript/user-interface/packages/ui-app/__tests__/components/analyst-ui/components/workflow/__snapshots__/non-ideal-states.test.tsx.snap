// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`workflow non-ideal-states workflow interval query empty matches snapshot 1`] = `
<BaseDisplay
  glContainer={
    Object {
      "close": [MockFunction],
      "emit": [MockFunction],
      "extendState": [MockFunction],
      "getElement": [MockFunction],
      "getState": [MockFunction],
      "height": 1200,
      "hide": [MockFunction],
      "isHidden": false,
      "layoutManager": undefined,
      "off": [MockFunction],
      "on": [MockFunction],
      "parent": undefined,
      "setSize": [MockFunction],
      "setState": [MockFunction],
      "setTitle": [MockFunction],
      "show": [MockFunction],
      "tab": undefined,
      "title": "workflow",
      "trigger": [MockFunction],
      "unbind": [MockFunction],
      "width": 1900,
    }
  }
>
  <div
    className="base-display "
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <WrappedWithNonIdealStateChecker
        operationalTimePeriodConfigurationQuery={
          Object {
            "canFetchMore": false,
            "clear": undefined,
            "data": undefined,
            "error": undefined,
            "failureCount": 0,
            "fetchMore": undefined,
            "isError": false,
            "isFetched": false,
            "isFetchedAfterMount": undefined,
            "isFetching": false,
            "isIdle": false,
            "isInitialData": undefined,
            "isLoading": false,
            "isPlaceholderData": undefined,
            "isPreviousData": undefined,
            "isStale": true,
            "isSuccess": true,
            "refetch": undefined,
            "remove": undefined,
            "status": undefined,
            "updatedAt": undefined,
          }
        }
        timeRange={
          Object {
            "endTimeSecs": 2000,
            "startTimeSecs": 1000,
          }
        }
        workflowIntervalQuery={
          Object {
            "canFetchMore": false,
            "clear": undefined,
            "data": undefined,
            "error": undefined,
            "failureCount": 0,
            "fetchMore": undefined,
            "isError": false,
            "isFetched": false,
            "isFetchedAfterMount": undefined,
            "isFetching": false,
            "isIdle": false,
            "isInitialData": undefined,
            "isLoading": false,
            "isPlaceholderData": undefined,
            "isPreviousData": undefined,
            "isStale": true,
            "isSuccess": true,
            "refetch": undefined,
            "remove": undefined,
            "status": undefined,
            "updatedAt": undefined,
          }
        }
        workflowQuery={
          Object {
            "canFetchMore": false,
            "clear": undefined,
            "data": undefined,
            "error": undefined,
            "failureCount": 0,
            "fetchMore": undefined,
            "isError": false,
            "isFetched": false,
            "isFetchedAfterMount": undefined,
            "isFetching": false,
            "isIdle": false,
            "isInitialData": undefined,
            "isLoading": false,
            "isPlaceholderData": undefined,
            "isPreviousData": undefined,
            "isStale": true,
            "isSuccess": true,
            "refetch": undefined,
            "remove": undefined,
            "status": undefined,
            "updatedAt": undefined,
          }
        }
      >
        <Blueprint3.NonIdealState
          description="Stage intervals returned empty"
          title="No data"
        >
          <div
            className="bp3-non-ideal-state"
          >
            <Component>
              <h4
                className="bp3-heading"
              >
                No data
              </h4>
            </Component>
            <div>
              Stage intervals returned empty
            </div>
          </div>
        </Blueprint3.NonIdealState>
      </WrappedWithNonIdealStateChecker>
    </Provider>
  </div>
</BaseDisplay>
`;

exports[`workflow non-ideal-states workflow interval query error matches snapshot 1`] = `
<BaseDisplay
  glContainer={
    Object {
      "close": [MockFunction],
      "emit": [MockFunction],
      "extendState": [MockFunction],
      "getElement": [MockFunction],
      "getState": [MockFunction],
      "height": 1200,
      "hide": [MockFunction],
      "isHidden": false,
      "layoutManager": undefined,
      "off": [MockFunction],
      "on": [MockFunction],
      "parent": undefined,
      "setSize": [MockFunction],
      "setState": [MockFunction],
      "setTitle": [MockFunction],
      "show": [MockFunction],
      "tab": undefined,
      "title": "workflow",
      "trigger": [MockFunction],
      "unbind": [MockFunction],
      "width": 1900,
    }
  }
>
  <div
    className="base-display "
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <WrappedWithNonIdealStateChecker
        operationalTimePeriodConfigurationQuery={
          Object {
            "canFetchMore": false,
            "clear": undefined,
            "data": undefined,
            "error": undefined,
            "failureCount": 0,
            "fetchMore": undefined,
            "isError": false,
            "isFetched": false,
            "isFetchedAfterMount": undefined,
            "isFetching": false,
            "isIdle": false,
            "isInitialData": undefined,
            "isLoading": false,
            "isPlaceholderData": undefined,
            "isPreviousData": undefined,
            "isStale": true,
            "isSuccess": true,
            "refetch": undefined,
            "remove": undefined,
            "status": undefined,
            "updatedAt": undefined,
          }
        }
        timeRange={
          Object {
            "endTimeSecs": 2000,
            "startTimeSecs": 1000,
          }
        }
        workflowIntervalQuery={
          Object {
            "canFetchMore": false,
            "clear": undefined,
            "data": undefined,
            "error": undefined,
            "failureCount": 0,
            "fetchMore": undefined,
            "isError": true,
            "isFetched": false,
            "isFetchedAfterMount": undefined,
            "isFetching": false,
            "isIdle": false,
            "isInitialData": undefined,
            "isLoading": false,
            "isPlaceholderData": undefined,
            "isPreviousData": undefined,
            "isStale": true,
            "isSuccess": true,
            "refetch": undefined,
            "remove": undefined,
            "status": undefined,
            "updatedAt": undefined,
          }
        }
        workflowQuery={
          Object {
            "canFetchMore": false,
            "clear": undefined,
            "data": undefined,
            "error": undefined,
            "failureCount": 0,
            "fetchMore": undefined,
            "isError": false,
            "isFetched": false,
            "isFetchedAfterMount": undefined,
            "isFetching": false,
            "isIdle": false,
            "isInitialData": undefined,
            "isLoading": false,
            "isPlaceholderData": undefined,
            "isPreviousData": undefined,
            "isStale": true,
            "isSuccess": true,
            "refetch": undefined,
            "remove": undefined,
            "status": undefined,
            "updatedAt": undefined,
          }
        }
      >
        <Blueprint3.NonIdealState
          description="Problem Loading Workflow Interval Data"
          title="Error"
        >
          <div
            className="bp3-non-ideal-state"
          >
            <Component>
              <h4
                className="bp3-heading"
              >
                Error
              </h4>
            </Component>
            <div>
              Problem Loading Workflow Interval Data
            </div>
          </div>
        </Blueprint3.NonIdealState>
      </WrappedWithNonIdealStateChecker>
    </Provider>
  </div>
</BaseDisplay>
`;

exports[`workflow non-ideal-states workflow interval query loading matches snapshot 1`] = `
<BaseDisplay
  glContainer={
    Object {
      "close": [MockFunction],
      "emit": [MockFunction],
      "extendState": [MockFunction],
      "getElement": [MockFunction],
      "getState": [MockFunction],
      "height": 1200,
      "hide": [MockFunction],
      "isHidden": false,
      "layoutManager": undefined,
      "off": [MockFunction],
      "on": [MockFunction],
      "parent": undefined,
      "setSize": [MockFunction],
      "setState": [MockFunction],
      "setTitle": [MockFunction],
      "show": [MockFunction],
      "tab": undefined,
      "title": "workflow",
      "trigger": [MockFunction],
      "unbind": [MockFunction],
      "width": 1900,
    }
  }
>
  <div
    className="base-display "
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <WrappedWithNonIdealStateChecker
        operationalTimePeriodConfigurationQuery={
          Object {
            "canFetchMore": false,
            "clear": undefined,
            "data": undefined,
            "error": undefined,
            "failureCount": 0,
            "fetchMore": undefined,
            "isError": false,
            "isFetched": false,
            "isFetchedAfterMount": undefined,
            "isFetching": false,
            "isIdle": false,
            "isInitialData": undefined,
            "isLoading": false,
            "isPlaceholderData": undefined,
            "isPreviousData": undefined,
            "isStale": true,
            "isSuccess": true,
            "refetch": undefined,
            "remove": undefined,
            "status": undefined,
            "updatedAt": undefined,
          }
        }
        timeRange={
          Object {
            "endTimeSecs": 2000,
            "startTimeSecs": 1000,
          }
        }
        workflowIntervalQuery={
          Object {
            "canFetchMore": false,
            "clear": undefined,
            "data": undefined,
            "error": undefined,
            "failureCount": 0,
            "fetchMore": undefined,
            "isError": false,
            "isFetched": false,
            "isFetchedAfterMount": undefined,
            "isFetching": false,
            "isIdle": false,
            "isInitialData": undefined,
            "isLoading": true,
            "isPlaceholderData": undefined,
            "isPreviousData": undefined,
            "isStale": true,
            "isSuccess": true,
            "refetch": undefined,
            "remove": undefined,
            "status": undefined,
            "updatedAt": undefined,
          }
        }
        workflowQuery={
          Object {
            "canFetchMore": false,
            "clear": undefined,
            "data": undefined,
            "error": undefined,
            "failureCount": 0,
            "fetchMore": undefined,
            "isError": false,
            "isFetched": false,
            "isFetchedAfterMount": undefined,
            "isFetching": false,
            "isIdle": false,
            "isInitialData": undefined,
            "isLoading": false,
            "isPlaceholderData": undefined,
            "isPreviousData": undefined,
            "isStale": true,
            "isSuccess": true,
            "refetch": undefined,
            "remove": undefined,
            "status": undefined,
            "updatedAt": undefined,
          }
        }
      >
        <Blueprint3.NonIdealState
          action={
            <Blueprint3.Spinner
              intent="primary"
            />
          }
          description="Workflow Interval Data"
          title="Loading"
        >
          <div
            className="bp3-non-ideal-state"
          >
            <Component>
              <h4
                className="bp3-heading"
              >
                Loading
              </h4>
            </Component>
            <div>
              Workflow Interval Data
            </div>
            <Blueprint3.Spinner
              intent="primary"
            >
              <div
                className="bp3-spinner bp3-intent-primary"
              >
                <div
                  className="bp3-spinner-animation"
                >
                  <svg
                    height={50}
                    strokeWidth="8.00"
                    viewBox="1.00 1.00 98.00 98.00"
                    width={50}
                  >
                    <path
                      className="bp3-spinner-track"
                      d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                    />
                    <path
                      className="bp3-spinner-head"
                      d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                      pathLength={280}
                      strokeDasharray="280 280"
                      strokeDashoffset={210}
                    />
                  </svg>
                </div>
              </div>
            </Blueprint3.Spinner>
          </div>
        </Blueprint3.NonIdealState>
      </WrappedWithNonIdealStateChecker>
    </Provider>
  </div>
</BaseDisplay>
`;

exports[`workflow non-ideal-states workflow query error matches snapshot 1`] = `
<BaseDisplay
  glContainer={
    Object {
      "close": [MockFunction],
      "emit": [MockFunction],
      "extendState": [MockFunction],
      "getElement": [MockFunction],
      "getState": [MockFunction],
      "height": 1200,
      "hide": [MockFunction],
      "isHidden": false,
      "layoutManager": undefined,
      "off": [MockFunction],
      "on": [MockFunction],
      "parent": undefined,
      "setSize": [MockFunction],
      "setState": [MockFunction],
      "setTitle": [MockFunction],
      "show": [MockFunction],
      "tab": undefined,
      "title": "workflow",
      "trigger": [MockFunction],
      "unbind": [MockFunction],
      "width": 1900,
    }
  }
>
  <div
    className="base-display "
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <WrappedWithNonIdealStateChecker
        operationalTimePeriodConfigurationQuery={
          Object {
            "canFetchMore": false,
            "clear": undefined,
            "data": undefined,
            "error": undefined,
            "failureCount": 0,
            "fetchMore": undefined,
            "isError": false,
            "isFetched": false,
            "isFetchedAfterMount": undefined,
            "isFetching": false,
            "isIdle": false,
            "isInitialData": undefined,
            "isLoading": false,
            "isPlaceholderData": undefined,
            "isPreviousData": undefined,
            "isStale": true,
            "isSuccess": true,
            "refetch": undefined,
            "remove": undefined,
            "status": undefined,
            "updatedAt": undefined,
          }
        }
        timeRange={
          Object {
            "endTimeSecs": 2000,
            "startTimeSecs": 1000,
          }
        }
        workflowIntervalQuery={
          Object {
            "canFetchMore": false,
            "clear": undefined,
            "data": undefined,
            "error": undefined,
            "failureCount": 0,
            "fetchMore": undefined,
            "isError": false,
            "isFetched": false,
            "isFetchedAfterMount": undefined,
            "isFetching": false,
            "isIdle": false,
            "isInitialData": undefined,
            "isLoading": false,
            "isPlaceholderData": undefined,
            "isPreviousData": undefined,
            "isStale": true,
            "isSuccess": true,
            "refetch": undefined,
            "remove": undefined,
            "status": undefined,
            "updatedAt": undefined,
          }
        }
        workflowQuery={
          Object {
            "canFetchMore": false,
            "clear": undefined,
            "data": undefined,
            "error": undefined,
            "failureCount": 0,
            "fetchMore": undefined,
            "isError": true,
            "isFetched": false,
            "isFetchedAfterMount": undefined,
            "isFetching": false,
            "isIdle": false,
            "isInitialData": undefined,
            "isLoading": false,
            "isPlaceholderData": undefined,
            "isPreviousData": undefined,
            "isStale": true,
            "isSuccess": true,
            "refetch": undefined,
            "remove": undefined,
            "status": undefined,
            "updatedAt": undefined,
          }
        }
      >
        <Blueprint3.NonIdealState
          action={
            <Blueprint3.Spinner
              intent="primary"
            />
          }
          description="Problem Loading Workflow Data"
          title="Error"
        >
          <div
            className="bp3-non-ideal-state"
          >
            <Component>
              <h4
                className="bp3-heading"
              >
                Error
              </h4>
            </Component>
            <div>
              Problem Loading Workflow Data
            </div>
            <Blueprint3.Spinner
              intent="primary"
            >
              <div
                className="bp3-spinner bp3-intent-primary"
              >
                <div
                  className="bp3-spinner-animation"
                >
                  <svg
                    height={50}
                    strokeWidth="8.00"
                    viewBox="1.00 1.00 98.00 98.00"
                    width={50}
                  >
                    <path
                      className="bp3-spinner-track"
                      d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                    />
                    <path
                      className="bp3-spinner-head"
                      d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                      pathLength={280}
                      strokeDasharray="280 280"
                      strokeDashoffset={210}
                    />
                  </svg>
                </div>
              </div>
            </Blueprint3.Spinner>
          </div>
        </Blueprint3.NonIdealState>
      </WrappedWithNonIdealStateChecker>
    </Provider>
  </div>
</BaseDisplay>
`;

exports[`workflow non-ideal-states workflow query loading matches snapshot 1`] = `
<BaseDisplay
  glContainer={
    Object {
      "close": [MockFunction],
      "emit": [MockFunction],
      "extendState": [MockFunction],
      "getElement": [MockFunction],
      "getState": [MockFunction],
      "height": 1200,
      "hide": [MockFunction],
      "isHidden": false,
      "layoutManager": undefined,
      "off": [MockFunction],
      "on": [MockFunction],
      "parent": undefined,
      "setSize": [MockFunction],
      "setState": [MockFunction],
      "setTitle": [MockFunction],
      "show": [MockFunction],
      "tab": undefined,
      "title": "workflow",
      "trigger": [MockFunction],
      "unbind": [MockFunction],
      "width": 1900,
    }
  }
>
  <div
    className="base-display "
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <WrappedWithNonIdealStateChecker
        operationalTimePeriodConfigurationQuery={
          Object {
            "canFetchMore": false,
            "clear": undefined,
            "data": undefined,
            "error": undefined,
            "failureCount": 0,
            "fetchMore": undefined,
            "isError": false,
            "isFetched": false,
            "isFetchedAfterMount": undefined,
            "isFetching": false,
            "isIdle": false,
            "isInitialData": undefined,
            "isLoading": false,
            "isPlaceholderData": undefined,
            "isPreviousData": undefined,
            "isStale": true,
            "isSuccess": true,
            "refetch": undefined,
            "remove": undefined,
            "status": undefined,
            "updatedAt": undefined,
          }
        }
        timeRange={
          Object {
            "endTimeSecs": 2000,
            "startTimeSecs": 1000,
          }
        }
        workflowIntervalQuery={
          Object {
            "canFetchMore": false,
            "clear": undefined,
            "data": undefined,
            "error": undefined,
            "failureCount": 0,
            "fetchMore": undefined,
            "isError": false,
            "isFetched": false,
            "isFetchedAfterMount": undefined,
            "isFetching": false,
            "isIdle": false,
            "isInitialData": undefined,
            "isLoading": false,
            "isPlaceholderData": undefined,
            "isPreviousData": undefined,
            "isStale": true,
            "isSuccess": true,
            "refetch": undefined,
            "remove": undefined,
            "status": undefined,
            "updatedAt": undefined,
          }
        }
        workflowQuery={
          Object {
            "canFetchMore": false,
            "clear": undefined,
            "data": undefined,
            "error": undefined,
            "failureCount": 0,
            "fetchMore": undefined,
            "isError": false,
            "isFetched": false,
            "isFetchedAfterMount": undefined,
            "isFetching": false,
            "isIdle": false,
            "isInitialData": undefined,
            "isLoading": true,
            "isPlaceholderData": undefined,
            "isPreviousData": undefined,
            "isStale": true,
            "isSuccess": true,
            "refetch": undefined,
            "remove": undefined,
            "status": undefined,
            "updatedAt": undefined,
          }
        }
      >
        <Blueprint3.NonIdealState
          action={
            <Blueprint3.Spinner
              intent="primary"
            />
          }
          description="Workflow Data"
          title="Loading"
        >
          <div
            className="bp3-non-ideal-state"
          >
            <Component>
              <h4
                className="bp3-heading"
              >
                Loading
              </h4>
            </Component>
            <div>
              Workflow Data
            </div>
            <Blueprint3.Spinner
              intent="primary"
            >
              <div
                className="bp3-spinner bp3-intent-primary"
              >
                <div
                  className="bp3-spinner-animation"
                >
                  <svg
                    height={50}
                    strokeWidth="8.00"
                    viewBox="1.00 1.00 98.00 98.00"
                    width={50}
                  >
                    <path
                      className="bp3-spinner-track"
                      d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                    />
                    <path
                      className="bp3-spinner-head"
                      d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                      pathLength={280}
                      strokeDasharray="280 280"
                      strokeDashoffset={210}
                    />
                  </svg>
                </div>
              </div>
            </Blueprint3.Spinner>
          </div>
        </Blueprint3.NonIdealState>
      </WrappedWithNonIdealStateChecker>
    </Provider>
  </div>
</BaseDisplay>
`;
